app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -144,6 +144,11 @@ class ScanSimpleIdUpdate @Inject constructor(
     if (model.isSearching) return noChange()
 
     val clearInvalidQrCodeModel = model.clearInvalidQrCodeError()
+
+    if (model.openedFrom == OpenedFrom.EditPatientScreen.ToAddNHID) {
+      return searchPatientWhenNHIDEnabled(clearInvalidQrCodeModel, event)
+    }
```

Comment:
This will be automatically done by the try/catch block. I don't think we don't need to handle it specifically. Is there a reason we are doing this? 🤔

Reply:
So the reason why we are doing it like this is because when we are opening the scanner to add bp passport, but actually scanning an NHID it should give an error. In the previous behaviour it would correctly scan the NHID as it would go inside the catch block which is what we wanted to avoid. Which is why if you even look at the test case we have written, we are testing if we scan BPPassport when we are supposed to scan NHID then it should output `InvalidQrCode` and not scan bp passport.

Reply:
Ah, got it. That makes sense, but we could've just handled the error states instead of breaking down the search functionality into different functions, right? For example, if the user scans an NHID when adding BP Passport, we can catch the error and check if it's opened to add the BP Passport and return early, no?
![image](https://user-images.githubusercontent.com/6140516/148153842-226d5848-9e6b-4fb7-b7a1-68efa71a8ad8.png)

Same, vice versa
![image](https://user-images.githubusercontent.com/6140516/148154005-91a83b19-af8b-4c7a-a28f-f7a4aff94f6e.png)

That way we just return early if the ID that is scanned is not what we want. Anyway, what we have works as well. Maybe we can make this change later on.